### PR TITLE
Dump/Restore Test

### PR DESCRIPTION
**Pull request by jgpruitt**

### CODE REVIEW

File: e2e/tests/dump-restore-test.rs [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+        exit.success(),
+        "copying the dump into the container failed: {}",
+        exit
+    );
+    let exit = Command::new("docker")
+        .arg("exec")
+        .arg(container_id)
+        .arg("psql")
+        .arg("-U")
+        .arg(username)
```
Comment (cevian) [inline e2e/tests/dump-restore-test.rs:233]: Nit: not sure if rustfmt allows this but would be nice to group args like this.
```suggestion
        .arg("-U").arg(username)
```

File: e2e/tests/dump-restore-test.rs [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    let dump = dump_db(container_id, "db0", "postgres"); // todo: dump using jack user
+    container.stop();
+    container.rm();
+
+    println!("dump file at {}", dump.display());
+
+    let container = run_postgres(&docker, "db1", "postgres");
+    let container_id = container.id();
+    psql_cmd(
+        container_id,
```
Comment (cevian) [inline e2e/tests/dump-restore-test.rs:280]: nit: are we sure it's better to pass in container ids rather than containers into these functions?

**Review by cevian — APPROVED** [review]
LGTM. Some nits, nothing major

File: e2e/tests/dump-restore-test.rs (hunk 1 above)
Comment (jgpruitt) [reply e2e/tests/dump-restore-test.rs:233]: I had the same thought and tried it, but it complained :(

File: e2e/tests/dump-restore-test.rs (hunk 2 above)
Comment (jgpruitt) [reply e2e/tests/dump-restore-test.rs:280]: no, not sure. the functions only need the id, but maybe the typing associated with passing the container itself would be nice? I'll give it a try...

Comment (jgpruitt) [reply e2e/tests/dump-restore-test.rs]: okay, that works well enough. changed it.

File: e2e/tests/dump-restore-test.rs (hunk 1 above)
Comment (jgpruitt) [reply e2e/tests/dump-restore-test.rs:233]: oh, but  I can do this: `.args(["-v", "ON_ERROR_STOP=1"])` :fire:

File: e2e/tests/dump-restore-test.rs [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "copying the dump out of the container failed: {}",
+        exit
+    );
+    path
```
[the remark this answers is not in the document]
Comment (jgpruitt) [reply e2e/tests/dump-restore-test.rs]: yup. on it.

[the remark this answers is not in the document]
Comment (jgpruitt) [reply e2e/tests/dump-restore-test.rs]: done.

**Pull request merged**